MonoGame.Framework/Platform/Graphics/Texture2D.OpenGL.cs

```diff
@@ -268,7 +268,7 @@ private void PlatformGetData<T>(int level, int arraySlice, Rectangle rect, T[] d
             {
                 // we need to convert from our format size to the size of T here
                 var tFullWidth = Math.Max(this.width >> level, 1) * Format.GetSize() / tSizeInByte;
-                var temp = new T[Math.Max(this.height >> level, 1) * tFullWidth];
+                var temp = GetDataPool<T>.GetArray(Math.Max(this.height >> level, 1) * tFullWidth);
```

Comment:
Are we gonna always leave behind a stale largest texture ever allocated?
i.e. it never gets deallocated unless the size mismatches correct on a subsequent call? This could be in MBs (size of screen).

Usually in a pool, you rent and then return. I don't see the return part here. In fact, why not just use `ArrayPool` like you said - it's not overkill imo?

Reply:
ArrayPool should be fine if `ArrayPool.Create()` is used with maxArrayLength = 4096 * 4096 (that should be enough) and maxArraysPerBucket = 1, since only one array will be used at a time

Reply:
Also maybe `maxArrayLength` should be `int.MaxValue` instead of `4096 * 4096`, so array can be any length and still be saved to the pool